### PR TITLE
[#2] [Chore] Configure Tests Automation

### LINKED ISSUE #2: [Chore] Configure Tests Automation
## Why

Test automation can help developer with catching bugs early on. It is also an oppotunity to learn CI for a Rails app.

## Acceptance Criteria

- Use Rails Template and Github Actions to set up unit test automation.
- Similarly, set up integration test automation.
- Test should perform and pass on the new branches.

### PR DESCRIPTION
**Pull request by blyscuit**
- Close #2

## What happened 👀

Run passing tests on new branches via GitHub Actions.

## Insight 📝

- Add generated `spec/spec_helper.rb`.
- Add `spec/support/simplecov.rb` for generating reports.
- Add `spec/string_spec.rb` for sample test.
- Add `spec/codebase/codebase_local_spec.rb` for running test from Danger locally.

## Proof Of Work 📹

Danger reports this PR's success.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Team
+ # @blyscuit is the sole developer
+ * @blyscuit @____ @____
```
Comment (malparty) [inline .github/CODEOWNERS]:
```suggestion
 * @blyscuit @byhbt @malparty
```

;-)

File: spec/codebase/codebase_spec.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Codebase', codebase: true do
+  it 'does not offend eslint' do
+    expect(`yarn run eslint`).to include 'Done'
+  end
```
Comment (malparty) [inline spec/codebase/codebase_spec.rb]:
```suggestion
  it 'does NOT offend Rubocop' do
    expect(`rubocop --parallel --format simple`).to include 'no offenses detected'
  end
```

ESLint is for linting JavaScript (EcmaScript), as this project is an API one, we might only need Rubocop.


Optionnaly you can also add Brakeman (for security checks) and Zeitwerk (just the Gem loader):

```

  it 'satisfies Brakeman' do
    expect(`brakeman -w1`).not_to include '+SECURITY WARNINGS+'
  end

  it 'does NOT break zeitwerk loading' do
    expect(`bundle exec rake zeitwerk:check`).to include 'All is good!'
  end
```

File: spec/systems/.keep
Comment (malparty) [inline spec/systems/.keep:1]: I think you might only need `request` tests and no need for `systems` (UI) tests in an API only project :) 
@byhbt let me know if you think differently 😇

Comment (blyscuit) [reply spec/systems/.keep:1]: Sounds good. I removed the system tests.

Comment (byhbt) [reply spec/systems/.keep:1]: yup, i think so, we just need the system test for the UI stuff: for example assert element on the UI. Or behavior of a user with a form.

File: spec/string_spec.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'String' do
+  context 'when converting to integer' do
+    it 'has correct value' do
+      expect('100'.to_i).to eq(100)
```
Comment (byhbt) [inline spec/string_spec.rb]: When writing a spec, we usually use `given... -> returns ....`. Just improve it for more specific.

```suggestion
  context 'given a string number' do
    it 'returns an integer' do
      expect('100'.to_i).to eq(100)
```

File: spec/codebase/codebase_local_spec.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  end
+
+  it 'satisfies Brakeman' do
+    expect(`brakeman -w1`).not_to include '+SECURITY WARNINGS+'
+  end
+
+  it 'does NOT break zeitwerk loading' do
+    expect(`bundle exec rake zeitwerk:check`).to be_empty.or include 'All is good!'
+  end
+end
```
Comment (malparty) [inline spec/codebase/codebase_local_spec.rb:17]: Just a personal sharing: I never felt the need to run these ones in the local env: as they provide only high-level warnings/recommendations, having them only in the PR war far enough to me (when there is a warning, I know I can refactor a minor thing, but at least I'm not spending time earlier running these many times).

It's personal so feel free to keep this `codebase_local` test file ;-)

**Review by malparty — APPROVED** [review]
Approved with a small note :)

File: spec/codebase/codebase_local_spec.rb (hunk 4 above)
Comment (blyscuit) [reply spec/codebase/codebase_local_spec.rb:17]: I included this in the local test to reflect what is being tested on CI and with our current set up, Danger will comment only on changed file while having it on local help dev see the bigger picture. I think it is useful for newer devs (like me). 🙇

File: spec/string_spec.rb (hunk 3 above)
Comment (blyscuit) [reply spec/string_spec.rb]: 👍 Will keep this in mind going forward.

**Review by byhbt — APPROVED** [review]

**Pull request merged**